src/main/java/build/buildfarm/worker/shard/Worker.java

```diff
@@ -226,10 +228,31 @@ private Server createServer(
           new WorkerProfileService(
               storage, inputFetchStage, executeActionStage, context, completeStage, backplane));
     }
+    handleGrpcMetricIntercepts(serverBuilder);
 
     return serverBuilder.build();
   }
 
+  public void handleGrpcMetricIntercepts(ServerBuilder<?> serverBuilder) {
```

Comment:
This could go into a common helper class to avoid duplication with the server.

Reply:
👍